bot/exts/moderation/silence.py

```diff
@@ -188,14 +330,25 @@ async def _unsilence(self, channel: TextChannel) -> bool:
             log.info(f"Tried to unsilence channel #{channel} ({channel.id}) but the channel was not silenced.")
             return False
 
-        overwrite = channel.overwrites_for(self._verified_role)
+        is_text_channel = isinstance(channel, TextChannel)
+
+        if is_text_channel:
+            overwrite = channel.overwrites_for(self._verified_msg_role)
+        else:
+            overwrite = channel.overwrites_for(self._verified_voice_role)
+
         if prev_overwrites is None:
             log.info(f"Missing previous overwrites for #{channel} ({channel.id}); defaulting to None.")
-            overwrite.update(send_messages=None, add_reactions=None)
+            overwrite.update(send_messages=None, add_reactions=None, speak=None, connect=None)
         else:
             overwrite.update(**json.loads(prev_overwrites))
 
-        await channel.set_permissions(self._verified_role, overwrite=overwrite)
+        if is_text_channel:
+            await channel.set_permissions(self._verified_msg_role, overwrite=overwrite)
+        else:
+            await channel.set_permissions(self._verified_voice_role, overwrite=overwrite)
+            await self._force_voice_sync(channel)
```

Comment:
Saving the result of `isinstance` is unnecessary since it's a cheap operation. The code could be simplified as follows:

```py
        if isinstance(channel, TextChannel):
            role = self._verified_msg_role
        else:
            role = self._verified_voice_role
        
        overwrite = channel.overwrites_for(role)

        if prev_overwrites is None:
            log.info(f"Missing previous overwrites for #{channel} ({channel.id}); defaulting to None.")
            overwrite.update(send_messages=None, add_reactions=None, speak=None, connect=None)
        else:
            overwrite.update(**json.loads(prev_overwrites))

        await channel.set_permissions(role, overwrite=overwrite)

        if isinstance(channel, VoiceChannel):
            await self._force_voice_sync(channel)
```